src/validation/coinbase.cairo

```diff
@@ -31,7 +33,17 @@ pub fn validate_coinbase(
     assert(total_output_amount <= total_fees + block_reward, 'total output > block rwd + fees');
 
     // TODO: validate BIP-34 sig script
-    // TODO: validate BIP-141 witness field
+
+    // validate BIP-141 witness field
+    if block_height >= BIP_141_BLOCK_HEIGHT {
+        let witness = tx.inputs[0].witness[0];
+
+        // check witness value
+        if witness != @"\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0" {
```

Comment:
Let's put it in a constant since it is a reserved value

`WITNESS_RESERVED_VALUE`

Reply:
I don't think ByteArray's are supported as constants.

Reply:
Fair enough